app/src/main/java/com/example/userinfo/UserInputActivity.kt

```diff
@@ -0,0 +1,164 @@
+            allFields.add(email)
+            allFields.add(phoneNumber)
+            allFields.add(pinCode)
+            allFields.add(address)
+            val allButtons=ArrayList<Button?>()
+            allButtons.add(confirmButton)
+            allButtons.add(validateButton)
+            allButtons.add(cancelButton)
+            val isAllFieldsChecked = checkAllFields(allFields)
+            if (isAllFieldsChecked) {
+                val isAllFieldsEnteredProperly = checkAllFieldsEnteredProperly(userName, email, phoneNumber, pinCode, address)
+                if (isAllFieldsEnteredProperly) {
+                    visible=false
+                    changeVisibilities(visible,userName,email,phoneNumber,pinCode,address,cancelButton,validateButton,confirmButton)
+                }
+            }
+        }
+        cancelButton?.setOnClickListener{
+            visible=true
+            changeVisibilities(visible,userName,email,phoneNumber,pinCode,address,cancelButton,validateButton,confirmButton)
+        }
+        confirmButton?.setOnClickListener{
+            val intent= Intent(this,DisplayUserInfo::class.java)
+            intent.putExtra(UserName,userName?.text.toString())
+            intent.putExtra(Address,address?.text.toString())
+            intent.putExtra(PinCode,pinCode?.text.toString())
+            intent.putExtra(PhoneNumber,phoneNumber?.text.toString())
+            intent.putExtra(Email,email?.text.toString())
+            startActivity(intent)
+        }
+    }
+
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        outState.putBoolean(Visible, visible!!)
+        super.onSaveInstanceState(outState)
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
+        super.onRestoreInstanceState(savedInstanceState)
+        visible=savedInstanceState.getBoolean(Visible)
+        val validateButton = findViewById<Button>(R.id.validateButton)
+        val userName = findViewById<EditText>(R.id.userNameET)
+        val email = findViewById<EditText>(R.id.emailET)
+        val phoneNumber = findViewById<EditText>(R.id.phoneNumberET)
+        val pinCode = findViewById<EditText>(R.id.pinCodeET)
+        val address = findViewById<EditText>(R.id.addressET)
+        val confirmButton= findViewById<Button>(R.id.ConfirmButton)
+        val cancelButton=findViewById<Button>(R.id.cancelButton)
+        changeVisibilities(visible,userName,email,phoneNumber,pinCode,address,cancelButton,validateButton,confirmButton)
+    }
+    private fun changeVisibilities(
+        visible: Boolean?,
+        userName: EditText,
+        email: EditText,
+        phoneNumber: EditText,
+        pinCode: EditText,
+        address: EditText,
+        cancelButton: Button,
+        validateButton: Button,
+        confirmButton: Button
+    ) {
+        confirmButton.visibility = if(!visible!!) View.VISIBLE else View.GONE
+        cancelButton.visibility = if(!visible) View.VISIBLE else View.GONE
```

Comment:
we can use Group when using ConstraintLayout to handle visibility of multiple( related) views
https://developer.android.com/reference/androidx/constraintlayout/widget/Group